mycli/main.py

```diff
@@ -1007,6 +1011,22 @@ def need_completion_refresh(queries):
         except Exception:
             return False
 
+
+def is_dropping_database(queries, dbname):
+    """Determine if the query is dropping a specific database."""
+    for query in sqlparse.split(queries):
+        try:
+            stmt = sqlparse.parse(query)[0]
+            first_token = stmt.token_first(skip_cm=True)
+            _, second_token = stmt.token_next(0, skip_cm=True)
+            if (first_token.value.lower() == 'drop' and
+                    second_token.value.lower() in ('database', 'schema') and
+                    stmt.get_name().lower() == dbname.lower()):
+                return True
+        except Exception:
```

Comment:
Hmm this is awful for debugging; can we use a more specific Exception or at least use:
```
            self.logger.debug('In is_dropp_database: %r.', e)                                                                          
            self.logger.error("traceback: %r", traceback.format_exc())                                                                       
```

BTW, I was thinking about adding a debug flag to disable the "catch all" exceptions in mycli (to help with debugging).